### PR TITLE
Add TabControl

### PR DESCRIPTION
**Pull request by 2yangk23**
This is just some parts of the TabControl which I believe should go in the framework. Will be used in osu for the filtering songs.

### CODE REVIEW

File: osu.Framework/Graphics/UserInterface/BasicDropDownHeader.cs [hunk 1]
```diff
@@ -0,0 +1,12 @@
+﻿// Copyright (c) 2007-2017 ppy Pty Ltd <contact@ppy.sh>.
+// Licensed under the MIT Licence - https://raw.githubusercontent.com/ppy/osu-framework/master/LICENCE
+
+using osu.Framework.Graphics.Containers;
+
+namespace osu.Framework.Graphics.UserInterface
+{
+    public abstract class BasicDropDownHeader : ClickableContainer
```
Comment (peppy) [inline osu.Framework/Graphics/UserInterface/BasicDropDownHeader.cs]: the idea behind Basic* classes is they are basic *implementations* of abstract classes, not the other way around.

**Comment by peppy** [comment]
- no test case for TabControl
- i'd probably remove the AtomicCounter and just make a local long in each tabcontrol instance (see [warning](https://ci.appveyor.com/project/peppy/osu-framework/build/1868/messages)).

File: osu.Framework/Graphics/UserInterface/BasicDropDownHeader.cs (hunk 1 above)
Comment (2yangk23) [inline osu.Framework/Graphics/UserInterface/BasicDropDownHeader.cs]: Found out there was a bug with Drawable when un-setting the RelativeSizeAxes. I fixed it now which should allow me to just extend DropDownHeader directly eliminating the need for this class.

**Comment by 2yangk23** [comment]
Currently there's no test case for TabControl since I don't know how it should be styled generically. Do we have guidelines on this? I've been testing in the osu tests meanwhile. The class is abstract for now so it forces users to extend all the components.

**Comment by peppy** [comment]
just make it functional, design is not important for framework

**Comment by peppy** [comment]
Currently there's no way to change the pinned tabs. I think we may need this in the future. What do you think about adding the pinned logic inside of `TabItem`?

**Comment by peppy** [comment]
Not sure how I feel about items being reordered when they are equally visible. I feel like this should only happen when selected from the dropdown (feels like more standard/user-accepted behaviour).

[Example](https://streamable.com/73zlp) (2 and 7 are pinned)

**Comment by 2yangk23** [comment]
Yea I agree it's confusing when that happens, I was basing the behavior off the specifications, but I will change it.

File: osu.Framework/Graphics/UserInterface/Tab/TabControl.cs [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+                    Direction = FillDirection.Full,
+                    RelativeSizeAxes = Axes.X,
+                    Height = dropDown.HeaderHeight,
+                    Masking = true,
+                    Padding = new MarginPadding
+                    {
+                        Left = offset,
+                        Right = dropDown.HeaderWidth
+                    },
+                    UpdateChild = updateDropDown,
```
Comment (peppy) [inline osu.Framework/Graphics/UserInterface/Tab/TabControl.cs]: rename `UpdateChild` to `TabVisibilityChanged`?

**Pull request merged**